docs/userguide/searchspace.md

```diff
@@ -1,46 +1,269 @@
+
+In BayBE, a search space is a union of two (potentially empty) subspaces. The [`SubspaceDiscrete`](baybe.searchspace.discrete.SubspaceDiscrete) contains all discrete parameters, while the [`SubspaceContinuous`](baybe.searchspace.continuous.SubspaceContinuous) contains all continuous parameters.
+
+Depending on which of the subspaces are non-empty, a `SearchSpace` has exactly one of the three [`SearchSpaceType`](baybe.searchspace.core.SearchSpaceType)'s:
+
+|`SubspaceDiscrete`|`SubspaceContinuous`|[`SearchSpaceType`](baybe.searchspace.core.SearchSpaceType)|
+|------------------|--------------------|-----------------|
+|Non-empty|Empty|[`SearchSpaceType.DISCRETE`](baybe.searchspace.core.SearchSpaceType.DISCRETE)|
+|Empty|Non-Empty|[`SearchSpaceType.CONTINUOUS`](baybe.searchspace.core.SearchSpaceType.CONTINUOUS)|
+|Non-Empty|Non-empty|[`SearchSpaceType.HYBRID`](baybe.searchspace.core.SearchSpaceType.HYBRID)|
+
+
+## `SubspaceDiscrete`
+
+The `SubspaceDiscrete` contains all the discrete parameters of a `SearchSpace`. There are different ways of constructing this subspace.
+
+### Building from the Product of Parameter Values
+
+The method [`SearchSpace.from_product`](baybe.searchspace.discrete.SubspaceDiscrete.from_product) constructs the full cartesian product of the provided parameters:
+
+```python
+from baybe.parameters import NumericalDiscreteParameter, CategoricalParameter
+from baybe.searchspace import SubspaceDiscrete
+
+parameters = [
+    NumericalDiscreteParameter(name="x0", values=[1, 2, 3]),
+    NumericalDiscreteParameter(name="x1", values=[4, 5, 6]),
+    CategoricalParameter(name="Speed", values=["slow", "normal", "fast"]),
+]
+subspace = SubspaceDiscrete.from_product(parameters=parameters)
+```
+
+In this example, `subspace` has a total of 27 different parameter configuration.
+
+~~~
+      x0   x1   Speed
+ 0   1.0  4.0    slow
+ 1   1.0  4.0  normal
+ 2   1.0  4.0    fast
+ ..  ...  ...     ...
+ 24  3.0  6.0    slow
+ 25  3.0  6.0  normal
+ 26  3.0  6.0    fast
+  
+  [27 rows x 3 columns]
+~~~
+
+### Constructing from a dataframe
+
+[`SubspaceDiscrete.from_dataframe`](baybe.searchspace.discrete.SubspaceDiscrete.from_dataframe) constructs a discrete subspace from a given dataframe.
+By default, this method tries to infer the data column as as a [`NumericalDiscreteParameter`](baybe.parameters.numerical.NumericalDiscreteParameter) and uses [`CategoricalParameter`](baybe.parameters.categorical.CategoricalParameter) as fallback.
+However, it is possible to change this behavior by using the optional `parameters` keyword.
+This list informs `from_dataframe` about the parameters and the types of parameters that should be used.
+In particular, it is necessary to provide such a list if there are non-numerical parameters that should not be interpreted as categorical parameters.
+
+```python
+import pandas as pd
+
+df = pd.DataFrame(
+    {
+        "x0": [2, 3],
+        "x1": [5, 4],
```

Comment:
this example and the corresponding print would be slightly more interesting if they had different amounts of unique values eg 
```python
        "x0": [2, 3, 3],
        "x1": [5, 4, 6],
...
```